juriscraper/opinions/united_states/state/nh.py

```diff
@@ -1,127 +1,150 @@
-        self.docket_name_pairs = self._get_anchor_docket_name_pairs()
-        return [pair["name"] for pair in self.docket_name_pairs]
-
-    def _get_download_urls(self):
-        path = "id('content')/div//ul//li//a[position()=1]"
-        download_url = []
-        for element in self.html.xpath(path):
-            url = element.xpath("./@href")[0]
-            download_url.extend([url])
-        return download_url
-
-    def _get_case_dates(self):
-        dates = []
-        path = 'id("content")/div//strong'
-        sub_path = (
-            "./following-sibling::ul[1]//li|../following-sibling::ul[1]//li"
-        )
-        for element in self.html.xpath(path):
-            text = element.text_content().strip().rstrip(":")
-            try:
-                case_date = convert_date_string(text)
-            except ValueError:
-                # Sometimes the court includes "To be released..."
-                # sections, without links, which we skip here
+    def _process_html(self) -> None:
+        for case in self.html["data"]:
+            content = fromstring(case["documentContent"])
+            urls = content.xpath(
+                ".//div[@class='document__detail__title']//a/@href"
+            )
+            if not urls:
+                logger.info(f"Opinion without URL to file: {case}")
                 continue
-            dates.extend([case_date] * len(element.xpath(sub_path)))
-        return dates
-
-    def _get_precedential_statuses(self):
-        return ["Published"] * len(self.case_names)
 
-    def _get_docket_numbers(self):
-        return [pair["docket"] for pair in self.docket_name_pairs]
+            date = case["documentPosted"]
+            if not date:
+                logger.info(f"Opinion without date: {case}")
+                continue
 
-    def _get_anchor_docket_name_pairs(self):
-        """The court has some ugly HTML practices that we need to handle.
+            name, docket = self.name_docket(case["documentName"])
+            self.cases.append(
+                {
+                    "name": name,
+                    "docket": docket,
+                    "date": date,
+                    "url": urls[0],
+                }
+            )
 
-        Most anchor links include single line strings with a single docket
-        number and a single case name.  However, there are two other formats
-        we've seen and must work around.
+    def name_docket(self, title) -> Tuple[str, str]:
+        """
+        Expected title values:
+            - 2020-0544 and 2020-0554, In re G.B.
+            - 2020-0528, 2020-0546, 2020-0548, In re S.A. & a.
+            - 2019-0283 In re R.M.
+            - 2017-0336 & 2019-0071  State of New Hampshire v. Daniel Turcotte
+            - 2014-0315, Deere & Company & a. v. The State of New Hampshire 2014-0441, Kubota Tractor Corporation v. The State of New Hampshire 2014-0575, Husqvarna Professional Products, Inc. v. The State of New Hampshire
```

Comment:
This method seems unnecessary.  And overly complicated.

Reply:
I would suggest deleting it and using my suggested code.

Reply:
Added your suggested code to the function.